### ISSUE TITLE
[feat] option to use the auto-mode.

### ISSUE THREAD

**Issue by rrterrorr** [swingerman/ha-dual-smart-thermostat#301]
is it posible to add auto-mode to dual thermostat.

>> set min, max and humidity

**Comment by swingerman** [comment]
Yes, but we need to define what behavior we expect in that mode.

**Comment by rrterrorr** [comment]
In auto mode, the behavior could work as follows:

1. **When heating is required:**  
   - The thermostat first checks if the current humidity is outside the acceptable range based on the dehumidification setpoint and its margin (e.g., if the setpoint is 50% and the current humidity is 61%, dehumidification is prioritized).  
   - Once the humidity is within 5% of the dehumidification setpoint (e.g., drops to 55% or below), the dehumidifier remains active to continue reaching the desired level, but heating may also start simultaneously to avoid delays.  
   - If the initial humidity is already within the acceptable range (e.g., 50-55%), the dehumidifier does not activate, and heating begins immediately.

2. **When cooling is required:**  
   - Humidity is not specifically addressed in this case, as air conditioners often handle dehumidification as part of their cooling process.

---

**Enhanced Automation in Auto Mode:**  
In addition to the above, the thermostat could use the configured setpoints for heating, cooling, and dehumidification—along with their respective margins—to determine the appropriate priorities dynamically. For example:

- **Heating Priority:**  
   If the room temperature is below the heating setpoint, the system checks if dehumidification is needed and adjusts priorities accordingly. Heating starts as soon as the humidity is within the acceptable range or close to it (within the margin).

- **Cooling Priority:**  
   If the room temperature exceeds the cooling setpoint, cooling is prioritized, with dehumidification handled simultaneously or as needed.

-**Dehumidification Priority:**
   If the humidity exceeds the dehumidification setpoint by more than the margin, the system prioritizes dehumidification, especially if the current mode is heating. Cooling can continue to handle dehumidification automatically when required.

By combining these priorities intelligently, the thermostat can efficiently balance heating, cooling, and dehumidification based on the current environmental conditions and the configured setpoints. This approach not only enhances user comfort but also optimizes energy usage.

**Comment by swingerman** [comment]
This is a great suggestion. I was thinking about what intelligence should be implemented. I was thinking about some predictions regarding pre-heating or pre-cooling based on presence or activity patterns, weather, day of the week, current season, wall and window isolation characteristics, heating, cooling speed data, etc. An ML model that learns and improves constantly.

**Comment by rrterrorr** [comment]
That sounds like an amazing idea! For additional inspiration or reference, you might want to check out this project: [HASmartThermostat on GitHub](https://github.com/ScratMan/HASmartThermostat).

**Comment by swingerman** [comment]
Tracking this as part of the comprehensive auto mode plan in #563. The approach: priority-based mode switching (not parallel operation, since many setups use virtual switches on a single AC unit). Also includes PID controller, outside temp compensation, and feels-like temperature as future phases.
